Applications/Spire/Source/Ui/ColorSelectorDropDown.cpp

```diff
@@ -143,10 +143,11 @@ ColorSelectorDropDown::ColorSelectorDropDown(const QColor& current_color,
   color_picker_layout->addWidget(m_color_hue_slider);
   m_hex_input = new ColorSelectorHexInputWidget(m_current_color, this);
   m_hex_input->setFixedSize(scale(122, 26));
-  m_hex_input->connect_color_signal([=] (const auto& color) {
-    m_color_value_slider->set_color(color);
-    m_color_hue_slider->set_color(color);
-    on_color_selected(color);
+  m_hex_input->connect_modified_signal([=] (const auto& color) {
+    on_color_hex_updated(color);
+  });
+  m_hex_input->connect_selected_signal([=] (const auto& color) {
+    on_color_hex_updated(color);
     window()->hide();
```

Comment:
Avoid having windows hide or show themselves. Sometimes it's necessary if the window is a top-level window but when you can avoid doing it it's preferable. Instead have the button hide the window when it receives a selected signal.

Reply:
I'll have to add a Changed signal to the ColorSelectorDropDown. The drop down is what differentiates between change and a selection signal, so it was more straightforward to close the window from there.